src/components/input.tsx

```diff
@@ -0,0 +1,64 @@
+import { type VariantProps, cva } from 'class-variance-authority'
+import React, { forwardRef } from 'react'
+
+import { cn } from '@/utils/cn'
+
+const inputVariants = cva(
+  [
+    'w-full h-14 rounded-xl border border-2 px-4 text-base text-medium',
+    'focus:outline-none focus:border-focus',
+    'placeholder-disabled',
+  ],
+
+  {
+    variants: {
+      variant: {
+        default: 'border-base',
+      },
+      error: {
+        true: 'border-error',
+      },
+      loading: {
+        true: 'text-transparent',
+      },
+      hasIcon: {
+        true: 'pl-12',
+      },
```

Comment:
If we need an icon, we will need to pass the following props:
```tsx
<Input hasIcon icon={<DollarSign />} {..restOfProps} />
```

Instead of having the `hasIcon` variant we can just apply the padding if the icon is present.
```suggestion
```